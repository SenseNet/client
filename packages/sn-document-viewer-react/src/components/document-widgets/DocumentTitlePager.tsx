import React, { useCallback, useState } from 'react'
import ClickAwayListener from '@material-ui/core/ClickAwayListener'
import TextField from '@material-ui/core/TextField'
import Typography from '@material-ui/core/Typography'
import { useDocumentData, useLocalization, useViewerState } from '../../hooks'

/**
 * Document widget component for paging
 */
export const DocumentTitlePager: React.FC = () => {
  const [isFocused, setIsFocused] = useState(false)

  const { documentData } = useDocumentData()
  const viewerState = useViewerState()
  const localization = useLocalization()

  const setPage = useCallback(
    (index: number) => {
      viewerState.updateState({ activePages: [index] })
      viewerState.onPageChange.setValue(index)
    },
    [viewerState],
  )

  const [currentPage, setCurrentPage] = useState(viewerState.activePages[0])

  const gotoPage = (page: string | number) => {
    let pageInt = typeof page === 'string' ? parseInt(page, 10) : page
    if (!isNaN(pageInt)) {
      pageInt = Math.max(pageInt, 1)
      pageInt = Math.min(pageInt, documentData.pageCount)
      setCurrentPage(pageInt)
      setPage(pageInt)
    }
  }

  return (
    <ClickAwayListener onClickAway={() => setIsFocused(false)}>
      <Typography
        onClick={() => setIsFocused(true)}
        variant="h6"
        color="inherit"
        style={{ cursor: 'pointer', display: 'flex', alignItems: 'center', overflow: 'hidden', margin: '0 2.5em' }}
        title={documentData.documentName}>
        <div style={{ whiteSpace: 'nowrap', overflow: 'hidden', textOverflow: 'ellipsis' }}>
          {documentData.documentName}&nbsp;
        </div>
        {isFocused ? (
          <form
            onSubmit={(ev) => {
              ev.preventDefault()
            }}>
            <TextField
              style={{ flexShrink: 0 }}
              title={localization.gotoPage}
              value={currentPage}
              onChange={(ev) => gotoPage(ev.currentTarget.value)}
              type="number"
              required={true}
              InputLabelProps={{
                shrink: true,
              }}
              inputProps={{
                min: 1,
                max: documentData.pageCount,
                style: { textAlign: 'center' },
              }}
              margin="dense"
            />
          </form>
        ) : (
          <div style={{ flexShrink: 0 }}>
            {currentPage} / {documentData.pageCount}
          </div>
        )}
      </Typography>
    </ClickAwayListener>
  )
}
