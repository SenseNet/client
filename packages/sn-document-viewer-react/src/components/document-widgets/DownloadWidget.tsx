import IconButton from '@material-ui/core/IconButton'
import CloudDownload from '@material-ui/icons/CloudDownload'
import React from 'react'
import { DocumentData } from '@sensenet/client-core'
import { useDocumentData, useLocalization } from '../../hooks'

/**
 * Own properties for the Share component
 */
export interface DownloadProps {
  download: (document: DocumentData) => void
}

/**
 * Component that allows active page rotation
 */
export const Download: React.FC<DownloadProps> = (props) => {
  const localization = useLocalization()
  const { documentData } = useDocumentData()
  return (
    <div style={{ display: 'inline-block' }}>
      <IconButton
        color="inherit"
        title={localization.download}
        onClick={() => props.download(documentData)}
        id="CloudDownload">
        <CloudDownload />
      </IconButton>
    </div>
  )
}
