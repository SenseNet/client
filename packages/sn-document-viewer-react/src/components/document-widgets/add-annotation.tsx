import { createStyles, makeStyles, Theme } from '@material-ui/core'
import { ChatSharp } from '@material-ui/icons'
import clsx from 'clsx'
import React from 'react'
import { useDocumentPermissions, useLocalization, useViewerState } from '../../hooks'
import { ToggleBase } from './toggle-base'

const useStyles = makeStyles((theme: Theme) => {
  return createStyles({
    iconButton: {},
    icon: {},
    iconActive: {
      fill: theme.palette.primary.main,
    },
  })
})

type AddAnnnotationClassKey = Partial<ReturnType<typeof useStyles>>

/**
 * Document widget component to draw Annotation
 */
export const AddAnnotationWidget: React.FC<{ classes?: AddAnnnotationClassKey }> = (props) => {
  const classes = useStyles(props)
  const localization = useLocalization()
  const viewerState = useViewerState()
  const permissions = useDocumentPermissions()

  const pageRotation =
    viewerState.rotation?.find((rotation) => rotation.pageNum === viewerState.activePage)?.degree || 0

  return (
    <ToggleBase
      disabled={!permissions.canEdit || pageRotation !== 0}
      classes={classes}
      active={viewerState.activeShapePlacing === 'annotation'}
      title={localization.addAnnotation}
      setValue={(value) => viewerState.updateState({ activeShapePlacing: value ? 'annotation' : 'none' })}>
      <ChatSharp
        className={clsx(classes.icon, { [classes.iconActive]: viewerState.activeShapePlacing === 'annotation' })}
      />
    </ToggleBase>
  )
}
