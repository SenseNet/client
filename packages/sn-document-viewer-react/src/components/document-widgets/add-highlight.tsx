import { createStyles, makeStyles, Theme } from '@material-ui/core'
import ChatBubbleOutlineSharp from '@material-ui/icons/ChatBubbleOutlineSharp'
import clsx from 'clsx'
import React from 'react'
import { useDocumentPermissions, useLocalization, useViewerState } from '../../hooks'
import { ToggleBase } from './toggle-base'

const useStyles = makeStyles((theme: Theme) => {
  return createStyles({
    iconButton: {},
    icon: {},
    iconActive: {
      fill: theme.palette.primary.main,
    },
  })
})

type AddRHightlightClassKey = Partial<ReturnType<typeof useStyles>>

/**
 * Document widget component to draw Hihgligh
 */
export const AddHighlightWidget: React.FC<{ classes?: AddRHightlightClassKey }> = (props) => {
  const classes = useStyles(props)
  const localization = useLocalization()
  const viewerState = useViewerState()
  const permissions = useDocumentPermissions()

  return (
    <ToggleBase
      disabled={!permissions.canEdit}
      classes={classes}
      isVisible={viewerState.isPlacingHighlight}
      title={localization.addHighlight}
      setValue={(value) => viewerState.updateState({ isPlacingHighlight: value })}>
      <ChatBubbleOutlineSharp
        className={clsx(classes.icon, { [classes.iconActive]: viewerState.isPlacingHighlight })}
      />
    </ToggleBase>
  )
}
