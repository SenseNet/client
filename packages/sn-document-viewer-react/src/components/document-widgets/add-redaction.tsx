import { createStyles, makeStyles, Theme } from '@material-ui/core'
import ChatBubbleSharp from '@material-ui/icons/ChatBubbleSharp'
import clsx from 'clsx'
import React from 'react'
import { useDocumentPermissions, useLocalization, useViewerState } from '../../hooks'
import { ToggleBase } from './toggle-base'

const useStyles = makeStyles((theme: Theme) => {
  return createStyles({
    iconButton: {},
    icon: {},
    iconActive: {
      fill: theme.palette.primary.main,
    },
  })
})

type AddRedactionClassKey = Partial<ReturnType<typeof useStyles>>

/**
 * Document widget component to toggleing redaction
 */
export const AddRedactionWidget: React.FC<{ classes?: AddRedactionClassKey }> = (props) => {
  const classes = useStyles(props)
  const localization = useLocalization()
  const viewerState = useViewerState()
  const permissions = useDocumentPermissions()

  return (
    <ToggleBase
      disabled={!permissions.canEdit}
      classes={classes}
      isVisible={viewerState.isPlacingRedaction}
      title={localization.addRedaction}
      setValue={(v) => viewerState.updateState({ isPlacingRedaction: v })}>
      <ChatBubbleSharp className={clsx(classes.icon, { [classes.iconActive]: viewerState.isPlacingRedaction })} />
    </ToggleBase>
  )
}
