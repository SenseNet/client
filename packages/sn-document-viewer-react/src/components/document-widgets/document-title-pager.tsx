import ClickAwayListener from '@material-ui/core/ClickAwayListener'
import TextField from '@material-ui/core/TextField'
import Typography from '@material-ui/core/Typography'
import React, { useState } from 'react'
import { useDocumentData, useLocalization, useViewerState } from '../../hooks'

/**
 * Document widget component for paging
 */
export const DocumentTitlePager: React.FC = () => {
  const { documentData } = useDocumentData()
  const viewerState = useViewerState()
  const localization = useLocalization()
  const [isFocused, setIsFocused] = useState(false)
  const [currentValue, setCurrentValue] = useState<number | string>(viewerState.activePage)

  const gotoPage = (page: string | number) => {
    let pageInt = typeof page === 'string' ? parseInt(page, 10) : page
    if (!isNaN(pageInt)) {
      pageInt = Math.max(pageInt, 1)
      pageInt = Math.min(pageInt, documentData.pageCount)
      viewerState.pageToGo.setValue({ page: pageInt })
    }
  }

  return (
    <ClickAwayListener onClickAway={() => setIsFocused(false)}>
      <Typography
        onClick={() => setIsFocused(true)}
        variant="h6"
        color="inherit"
        style={{ cursor: 'pointer', display: 'flex', alignItems: 'center', overflow: 'hidden', margin: '0 2.5em' }}
        title={documentData.documentName}>
        <div style={{ whiteSpace: 'nowrap', overflow: 'hidden', textOverflow: 'ellipsis' }}>
          {documentData.documentName}&nbsp;
        </div>
        {isFocused ? (
          <form
            onSubmit={(ev) => {
              gotoPage(currentValue)
              ev.preventDefault()
            }}>
            <TextField
              style={{ flexShrink: 0 }}
              title={localization.gotoPage}
              onChange={(ev) => setCurrentValue(ev.currentTarget.value)}
              onBlur={() => gotoPage(currentValue)}
              defaultValue={viewerState.activePage}
              type="number"
              required={true}
              InputLabelProps={{
                shrink: true,
              }}
              inputProps={{
                min: 1,
                max: documentData.pageCount,
                style: { textAlign: 'center' },
              }}
              margin="dense"
            />
          </form>
        ) : (
          <div style={{ flexShrink: 0 }}>
            {viewerState.activePage} / {documentData.pageCount}
          </div>
        )}
      </Typography>
    </ClickAwayListener>
  )
}
