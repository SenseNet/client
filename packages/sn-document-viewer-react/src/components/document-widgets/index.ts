export * from './add-annotation'
export * from './add-highlight'
export * from './add-redaction'
export * from './document-title-pager'
export * from './download-widget'
export * from './print-widget'
export * from './rotate-active-pages'
export * from './rotate-document'
export * from './rotate-widget'
export * from './save-widget'
export * from './share-widget'
export * from './toggle-base'
export * from './toggle-comments'
export * from './toggle-redaction'
export * from './toggle-shapes'
export * from './toggle-thumbnails'
export * from './toggle-watermark'
export * from './zoom-in-out'
