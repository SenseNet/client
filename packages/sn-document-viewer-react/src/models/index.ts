export * from './DocumentViewerApiSettings'
export * from './Theming'
export * from './viewer-state'
