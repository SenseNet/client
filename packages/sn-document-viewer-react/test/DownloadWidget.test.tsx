import { shallow } from 'enzyme'
import React from 'react'
import { DownloadComponent } from '../src/components/document-widgets/DownloadWidget'
import { DocumentData } from '../src/models/DocumentData'

describe('DownloadWidget component', () => {
  it('Should render without crashing', () => {
    const wrapper = shallow(<DownloadComponent download={jest.fn()} title="" document={{} as DocumentData} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('Should trigger a download request when clicked', () => {
    const download = jest.fn()
    const wrapper = shallow(<DownloadComponent download={download} title="" document={{} as DocumentData} />)
    wrapper.find('#CloudDownload').simulate('click')
    expect(download).toBeCalled()
  })
})
