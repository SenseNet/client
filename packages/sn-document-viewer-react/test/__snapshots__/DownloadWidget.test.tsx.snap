// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownloadWidget component Should render without crashing 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
    }
  }
>
  <WithStyles(IconButton)
    color="inherit"
    id="CloudDownload"
    onClick={[Function]}
    title=""
  >
    <pure(CloudDownloadIcon) />
  </WithStyles(IconButton)>
</div>
`;
